eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/DependencyAnalyzer.java

```diff
@@ -0,0 +1,207 @@
+        }
+    }
+
+    public TreeSet<BomDependency> analyze() {
+        resolveTree();
+        filterConflicts();
+
+        return this.bomEligibleDependencies;
+    }
+
+    public Boolean validate() {
+        resolveTree();
+        return filterConflicts();
+    }
+
+    private void resolveTree() {
+        for (MavenDependency gaLibrary : inputDependencies) {
+            try {
+                MavenResolvedArtifact mavenResolvedArtifact = null;
+
+                mavenResolvedArtifact = getMavenResolver()
+                    .addDependency(gaLibrary)
+                    .resolve()
+                    .withoutTransitivity()
+                    .asSingleResolvedArtifact();
+
+                BomDependency parentDependency = new BomDependency(mavenResolvedArtifact.getCoordinate());
+                MavenArtifactInfo[] dependencies = mavenResolvedArtifact.getDependencies();
+
+                addDependencyToDependencyTree(parentDependency, null, nameToVersionToChildrenDependencyTree);
+
+                for (MavenArtifactInfo dependency : dependencies) {
+                    if (dependency.getScope() == ScopeType.TEST) {
+                        continue;
+                    }
+                    if(RESOLVED_EXCLUSION_LIST.contains(dependency.getCoordinate().getArtifactId())) {
+                        // This is coming from the BOM, we may not include this.
+                        continue;
+                    }
+
+                    BomDependency childDependency = new BomDependency(dependency.getCoordinate());
+                    addDependencyToDependencyTree(childDependency, parentDependency, nameToVersionToChildrenDependencyTree);
+                }
+            }
+            catch(Exception ex) {
+                System.out.println(ex);
+            }
+        }
+    }
+
+    private static MavenResolverSystemBase<PomEquippedResolveStage, PomlessResolveStage, MavenStrategyStage, MavenFormatStage> getMavenResolver() {
+        return Maven.configureResolver().withMavenCentralRepo(true);
+    }
+
+    private static void addDependencyToDependencyTree(BomDependency dependency, BomDependency parentDependency, TreeMap<BomDependencyNoVersion, HashMap<String, Collection<BomDependency>>> dependencyTree) {
+        if (!dependencyTree.containsKey(dependency)) {
+            dependencyTree.put(new BomDependencyNoVersion(dependency.getGroupId(), dependency.getArtifactId()), new HashMap<>());
+        }
+
+        HashMap<String, Collection<BomDependency>> versionToParents = dependencyTree.get(dependency);
+        if(!versionToParents.containsKey(dependency.getVersion())) {
+            versionToParents.put(dependency.getVersion(), new ArrayList<>());
+        }
+
+        if(parentDependency != null) {
+            versionToParents.get(dependency.getVersion()).add(parentDependency);
+        }
+    }
+
+
+    private void makeDependencyInEligible(BomDependency dependency) {
```

Comment:
Add logs to indicate why dependencies are marked as ineligible.

Reply:
Yes, I have a work-item to add better logging\reporting in the system.